uniffi_bindgen/src/lib.rs

```diff
@@ -242,18 +246,19 @@ pub trait BindingGenerator: Sized {
 /// - `udl_file`: The path to the UDL file
 /// - `config_file_override`: The path to the configuration toml file, most likely called `uniffi.toml`. If [`None`], the function will try to guess based on the crate's root.
 /// - `out_dir_override`: The path to write the bindings to. If [`None`], it will be the path to the parent directory of the `udl_file`
-pub fn generate_external_bindings<B: BindingGenerator, P: AsRef<Path>>(
-    udl_file: P,
-    config_file_override: Option<P>,
-    out_dir_override: Option<P>,
+pub fn generate_external_bindings(
+    binding_generator: impl BindingGenerator,
+    udl_file: impl AsRef<Path>,
```

Comment:
just out of curiosity, should we change this to be `&Path` like we did in `write_bindings`? (I know it was a generic parameter before this change, so no need for any changes here 😄)

Reply:
We could, but to me this feels like the right function to put `as_ref()` call.  I don't really have any strong reason for it though, so I could be easily convinced otherwise.